cephfs/statx.go

```diff
@@ -0,0 +1,133 @@
+	// traditional stat call.
+	StatxBasicStats = StatxWant(C.CEPH_STATX_BASIC_STATS)
+	// StatxBtime requests the birth-time value be filled in.
+	StatxBtime = StatxWant(C.CEPH_STATX_BTIME)
+	// StatxVersion requests the version value be filled in.
+	StatxVersion = StatxWant(C.CEPH_STATX_VERSION)
+	// StatxAllStats requests all known stat values be filled in.
+	StatxAllStats = StatxWant(C.CEPH_STATX_ALL_STATS)
+)
+
+// AtFlags represent flags to be passed to calls that control how files
+// are used or referenced. For example, not following symlinks.
+type AtFlags uint
+
+const (
+	// AtNoAttrSync requests that the stat call only fetch locally-cached
+	// values if possible, avoiding round trips to a back-end server.
+	AtNoAttrSync = AtFlags(C.AT_NO_ATTR_SYNC)
+	// AtSymlinkNofollow indicates the call should not follow symlinks
+	// but operate on the symlink itself.
+	AtSymlinkNofollow = AtFlags(C.AT_SYMLINK_NOFOLLOW)
+)
+
+// CephStatx instances are returned by extended stat (statx) calls.
+// Note that CephStatx results are similar to but not identical
+// to (Linux) system statx results.
+type CephStatx struct {
+	statx C.struct_ceph_statx
+}
+
+// Mask returns a value containing the bits indicating what
+// statx fields have been set.
+func (csx *CephStatx) Mask() StatxWant {
+	return StatxWant(csx.statx.stx_mask)
+}
+
+// Blksize returns the block size.
+func (csx *CephStatx) Blksize() uint32 {
+	return uint32(csx.statx.stx_blksize)
+}
+
+// Nlink returns the number of links for the file.
+func (csx *CephStatx) Nlink() uint32 {
+	return uint32(csx.statx.stx_nlink)
+}
+
+// Uid returns the file owner's UID.
+func (csx *CephStatx) Uid() uint32 {
+	return uint32(csx.statx.stx_uid)
+}
+
+// Gid returns the file owner's GID.
+func (csx *CephStatx) Gid() uint32 {
+	return uint32(csx.statx.stx_gid)
+}
+
+// Mode return the file's type and mode value.
+func (csx *CephStatx) Mode() uint16 {
+	return uint16(csx.statx.stx_mode)
+}
+
+// Inode returns the inode number of the file.
+func (csx *CephStatx) Inode() Inode {
+	return Inode(csx.statx.stx_ino)
+}
+
+// Size returns the size of the file in bytes.
+func (csx *CephStatx) Size() uint64 {
+	return uint64(csx.statx.stx_size)
+}
+
+// Blocks returns the number of blocks allocated for the file.
+func (csx *CephStatx) Blocks() uint64 {
+	return uint64(csx.statx.stx_blocks)
+}
+
+// Dev returns the device id for the device containing the file system.
+func (csx *CephStatx) Dev() uint64 {
+	return uint64(csx.statx.stx_dev)
+}
+
+// Rdev returns the device id if the current file represents a device.
+func (csx *CephStatx) Rdev() uint64 {
+	return uint64(csx.statx.stx_rdev)
+}
+
+// Version returns a file version value.
+func (csx *CephStatx) Version() uint64 {
+	return uint64(csx.statx.stx_version)
+}
+
+// TODO: missing timespec fields:
+//  Do we define our own Timespec type or use one from
+//   https://godoc.org/golang.org/x/sys/unix#Timespec
+//  or is there some other option?
```

Comment:
I would not like to use our own Timespec, use the one from x/sys. If there are concerns later on, we can always decide to implement our own, or use something else.

Reply:
Doing a naive version of this and then deciding later to change it could be a breaking api change. This wouldn't be a big deal in v0.y but ought to be considered a no-no post v1.y. However I do have one idea that may split the difference, defining a `type Timespec unix.Timespec`. This allows us to re-use the existing implmentation, add our own methods if needed, etc w/o promising we'll always return a unix.Timespec. Thoughts?

Reply:
yeah, that is a nice approach.